.ado/templates/apple-job-react-native.yml

```diff
@@ -69,6 +69,16 @@ steps:
       xcode_actions: ${{ parameters.xcode_actions_release }}
       xcode_useXcpretty: false
       xcode_destination: ${{ parameters.xcode_destination }}
+  
+  - template: apple-xcode-build.yml
+    parameters:
+      xcode_sdk: ${{ parameters.xcode_sdk }}
+      xcode_configuration: Debug -enableAddressSanitizer YES
```

Comment:
Could we just do through an xcconfig so the CI matches the experience when you build locally in xcode? As is this means address sanitizer issues could be exposed through CI but then it wouldn't be immediately obvious to the engineer investigating how they could repro the issue

Reply:
I could make a new scheme + xcconfig pairing just for ASAN builds, but the downside is that's taking on another big diff from upstream.

I'll point out that by copying the Xcodebuild command from the CI I was able to run and test this change easily from command line locally.

Reply:
If you don't have separate xcconfigs, can just add scoped settings like this in your main xcconfig:
ENABLE_ADDRESS_SANITIZER[config=Debug] = YES
ENABLE_ADDRESS_SANITIZER[config=Release] = NO

Reply:
But then every debug build will have ASAN on? That'd increase everyone's build times, even locally. I'll run some diagnostics to see how much longer it takes to run with ASAN locally to get a better idea on how much doing that will disrupt one's workflow.

Reply:
A pro to adding it just to the CI was it won't slow down everyone's builds locally...

Reply:
What's the actual build time impact? I think for actual security we probably want this running in dev environment as well as CI so impacting everyone's builds would be a good thing. See also the comment from @harrieshin about wanting this on for everyone using RNTester.

And then historically for perceived build reliability we've just found the more settings that are different between default and CI/publish the more confusing it is for people. Of course you can look at the build logs and figure out which flags are applicable/able to run locally and hopefully repro, but that's a lot of effort when someone is probably already frustrated because what worked for them locally is failing in CI.